lib/iris/cube.py

```diff
@@ -880,8 +880,11 @@ def convert_units(self, unit):
 
         """
         # If the cube has units convert the data.
-        if not self.units.is_unknown():
-            self.data = self.units.convert(self.data, unit)
+        if self.units.is_unknown():
+            raise iris.exceptions.UnitConversionError(
+                'cannot convert from unknown units to {!s}, '
+                'set the "units" attribute instead'.format(unit))
```

Comment:
I think the second line of the error message may confuse rather than help, so perhaps we can drop it. I'd also suggest printing the repr of the unit's name in the error message; that is:

```python
"cannot convert from 'unknown' units to {!r}.".format(unit.origin)
```